From 81ca06557c6071c3e64556123ac6f6afd0af4d42 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Tue, 16 Aug 2022 13:53:28 +0200
Subject: [PATCH] ARM: dts: qcom: ipq8064: fix smb208 regulator node

Fix wrong switch-mode-frequency binding. The smb208 regulators support
only 500Hz or 1Mhz, fix the wrong value to 500Hz following original QSDK
code.
Add missing system-load and bias-pull-down binding for smb208
regulators. These binding were never added and the values comes from
ancient QSDK board data.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 arch/arm/boot/dts/qcom-ipq8062-smb208.dtsi      | 16 ++++++++++++----
 arch/arm/boot/dts/qcom-ipq8064-smb208.dtsi      | 16 ++++++++++++----
 arch/arm/boot/dts/qcom-ipq8064-v2.0-smb208.dtsi | 16 ++++++++++++----
 arch/arm/boot/dts/qcom-ipq8065-smb208.dtsi      | 16 ++++++++++++----
 4 files changed, 48 insertions(+), 16 deletions(-)

--- a/arch/arm/boot/dts/qcom-ipq8062-smb208.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8062-smb208.dtsi
@@ -9,29 +9,41 @@
 		smb208_s1a: s1a {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1200000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s1b: s1b {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1679000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2a: s2a {
 			regulator-min-microvolt = < 800000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2b: s2b {
 			regulator-min-microvolt = < 800000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 	};
 };
--- a/arch/arm/boot/dts/qcom-ipq8064-smb208.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8064-smb208.dtsi
@@ -9,29 +9,41 @@
 		smb208_s1a: s1a {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1200000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s1b: s1b {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1679000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2a: s2a {
 			regulator-min-microvolt = < 800000>;
 			regulator-max-microvolt = <1250000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2b: s2b {
 			regulator-min-microvolt = < 800000>;
 			regulator-max-microvolt = <1250000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 	};
 };
--- a/arch/arm/boot/dts/qcom-ipq8064-v2.0-smb208.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8064-v2.0-smb208.dtsi
@@ -9,29 +9,41 @@
 		smb208_s1a: s1a {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1200000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s1b: s1b {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1679000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2a: s2a {
 			regulator-min-microvolt = < 800000>;
 			regulator-max-microvolt = <1250000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2b: s2b {
 			regulator-min-microvolt = < 800000>;
 			regulator-max-microvolt = <1250000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 	};
 };
--- a/arch/arm/boot/dts/qcom-ipq8065-smb208.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq8065-smb208.dtsi
@@ -9,29 +9,41 @@
 		smb208_s1a: s1a {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1200000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s1b: s1b {
 			regulator-min-microvolt = <1050000>;
 			regulator-max-microvolt = <1150000>;
+			regulator-system-load = <1679000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2a: s2a {
 			regulator-min-microvolt = <775000>;
 			regulator-max-microvolt = <1275000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 
 		smb208_s2b: s2b {
 			regulator-min-microvolt = <775000>;
 			regulator-max-microvolt = <1275000>;
+			regulator-system-load = <1740000>;
+			regulator-settling-time-us = <10>;
 
-			qcom,switch-mode-frequency = <1200000>;
+			qcom,switch-mode-frequency = <500000>;
+			bias-pull-down;
 		};
 	};
 };
